layout/layout.c

```diff
@@ -806,48 +806,45 @@ M0_INTERNAL void m0_layout_enum_type_unregister(struct m0_layout_domain *dom,
 	M0_LEAVE("Enum_type_id %lu", (unsigned long)let->let_id);
 }
 
-M0_INTERNAL uint64_t m0_layout_find_by_buffsize(struct m0_layout_domain *dom,
-						struct m0_fid *pver,
-						size_t buffsize)
+M0_INTERNAL int m0_layout_find_by_buffsize(struct m0_layout_domain *dom,
+					   struct m0_fid *pver,
+					   size_t buffsize)
 {
-	uint64_t lid = M0_DEFAULT_LAYOUT_ID;
-	struct m0_pdclust_attr *pa;
-	struct m0_layout *l;
-	uint64_t hash;
+	struct m0_pdclust_attr *pa = NULL;
+	struct m0_layout *l = NULL;
+	uint64_t hash = 0;
 	int i;
 
 	m0_mutex_lock(&dom->ld_lock);
 	for (i = M0_DEFAULT_LAYOUT_ID; i < m0_lid_to_unit_map_nr; ++i) {
-		/* Current BE max tx size constraints. */
-		if (m0_lid_to_unit_map[i] > 4 * LNET_MAX_PAYLOAD)
-			break;
 		hash = m0_pool_version2layout_id(pver, i);
 		l = m0_layout__list_lookup(dom, hash, true);
 		if (l != NULL) {
 			pa = &m0_layout_to_pdl(l)->pl_attr;
-			if (pa->pa_unit_size * pa->pa_N <= buffsize) {
-				lid = i;
-			} else {
-				m0_ref_put(&l->l_ref);
+			if (pa->pa_unit_size * pa->pa_N >= buffsize ||
+			    /*
+			     * Performance degrades for units bigger than
+			     * 4 * LNET_MAX_PAYLOAD (4MB atm), as tests show.
```

Comment:
Could you explain why?

Reply:
As it is written: "as tests show". Why is it so? My guess is because as the number of 1MB lnet frames to be transferred in one io fom increases, the time for BE tx to be closes also increases. And this, in turn, affects the overall performance of transaction processing. But this is only my guess...